lib/websocket/connection.js

```diff
@@ -256,7 +256,10 @@ function socketClosed (ws) {
   const { [kResponse]: response } = ws
 
   response.socket.on('close', () => {
-    const wasClean = ws[kReadyState] === states.CLOSING || isEstablished(ws)
+    // If the TCP connection was closed after the
+    // WebSocket closing handshake was completed, the WebSocket connection
+    // is said to have been closed _cleanly_.
+    const wasClean = ws[kSentClose]
```

Comment:
You are probably already aware of this but receiving the socket `'close'` event does not mean that the closing handshake was completed.

Reply:
I was on the fence about where setting this was. Thinking it over, it should have been set after the server receives the close frame.